Authentication/SignUp/View/SignupView.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  SignupViewType.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/18/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+
+public protocol SignupViewType: Renderable {
+ 
+    var view: UIView { get }
+    
+    var usernameLabel: UILabel? { get }
+    var usernameTextField: UITextField? { get }
+    var usernameValidationMessageLabel: UILabel? { get }
+    
+    var emailLabel: UILabel? { get }
+    var emailTextField: UITextField { get }
+    var emailValidationMessageLabel: UILabel? { get }
+    
+    var passwordLabel: UILabel? { get }
+    var passwordTextField: UITextField { get }
+    var passwordValidationMessageLabel: UILabel? { get }
+    var passwordVisibilityButton: UIButton? { get }
+    
+    var passwordConfirmLabel: UILabel? { get }
+    var passwordConfirmTextField: UITextField { get }
+    var passwordConfirmValidationMessageLabel: UILabel? { get }
+    var passwordConfirmVisibilityButton: UIButton? { get }
+    
+    var signupButton: UIButton { get }
+    var signupErrorLabel: UILabel? { get }
+    
+    var termsAndServicesButton: UIButton { get }
+    var termsAndServicesLabel: UILabel? { get }
+    
+    var usernameTextFieldValid: Bool { get set }
+    var usernameTextFieldSelected: Bool { get set }
+    var emailTextFieldValid: Bool { get set }
+    var emailTextFieldSelected: Bool { get set }
+    var passwordTextFieldValid: Bool { get set }
+    var passwordTextFieldSelected: Bool { get set }
+    var passwordConfirmationTextFieldValid: Bool { get set }
+    var passwordConfirmationTextFieldSelected: Bool { get set }
+    var signupButtonEnabled: Bool { get set }
+    var signupButtonPressed: Bool { get set }
+    
```

Comment:
I see `SignUpFormType` and `LoginFormType` both as an extension of the `AuthenticationFormType`, and by doing that, we assure that we are consistent for the things we are asking for in the `AuthenticationFormType`. For example, if we need an email for login, we are sure that we are also asking for it in `AuthenticationFormType`. I'm just saying this because I just realized that we gain by having both extend a common protocol.


For the `Renderable`, first I thought that it didn't need to extend `Renderable`. Now that I see it, both ```SignUpFormType: AuthenticationFormType, Renderable``` and ```typealias SignUpViewType = protocol<SignUpFormType, Renderable>``` are the same. The only thing is that (I don't know why) I assumed that the second way of writing gives a better notion of composability than the first one, because one (which happened to me) can see it as inheritance. But they are the same and it could be written both ways, maybe the first one is better so we avoid that extra line.